recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+
+    @property
+    def _cmake_cpp_standard(self):
+        return ''.join(c for c in str(self.settings.compiler.cppstd) if c in string.digits)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "aws-sdk-cpp-{}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        if not self.settings.compiler.cppstd:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = 11
+        self._cmake.definitions["BUILD_DEPS"] = False
+        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
+        self._cmake.definitions["ENABLE_RTTI"] = self.options.with_rtti
+        self._cmake.definitions["FORCE_CURL"] = self.options.with_curl
+        self._cmake.definitions["ENABLE_CURL_LOGGING"] = self.options.get_safe("enable_curl_logging")
+        if self.settings.compiler.cppstd:
+            self._cmake.definitions["CPP_STANDARD"] = self._cmake_cpp_standard
+
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["SIMPLE_INSTALL"] = True
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in str(self.settings.compiler.runtime)
+
+        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
+
+        self._cmake.definitions["ANDROID_BUILD_CURL"] = False
+        self._cmake.definitions["ANDROID_BUILD_OPENSSL"] = False
+        self._cmake.definitions["ANDROID_BUILD_ZLIB"] = False
+
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "sdks.cmake"),
+                              "sort_links(EXPORTS)", "# sort_links(EXPORTS)")
+        for root, _, _ in os.walk(self._source_subfolder):
+            cmakelists = os.path.join(root, "CMakeLists.txt")
+            repl = "AWS::aws-c-event-stream"
+            if os.path.isfile(cmakelists) and repl in tools.load(cmakelists):
+                tools.replace_in_file(cmakelists, repl, "CONAN_PKG::aws-c-event-stream", strict=False)
+
+    @property
+    def _cpu_count(self):
+        try:
+            cpus = int(tools.cpu_count())
+            return min((cpus, 2))
+        except ValueError:
+            return 2
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        # Reduce parallelism to avoid out-of-memory on limited memory build systems
+        with tools.environment_append({"CONAN_CPU_COUNT": str(self._cpu_count)}):
+            cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

Comment:
why delete this? I, as a consumer, prefer the library's cmake config script, if available, rather than conan's generated one

Reply:
It's cci policy to remove these.
But I agree to your sentiment.
Most cmake config scripts now are relocatable, whereas in the past they had absolute paths to the build systems embedded into them.


@danimtb @uilianries @SpaceIm @jgsogo 
aws-sdk-cpp provides a lot of libraries (and config cmake scripts). By a lot, I mean 30+.
conan does currently not provide a way to generate multiple config scripts.
And even then, the generated conan cmake scripts will be generic and never the same as those provided by the aws-sdk-cpp cmake build script.

I feel cci should allow those cmake scripts that play by the book and are fully relocatable.
This would also save a lot of maintenance and acceptance of conan/cci.

Reply:
It seems that the other AWS packages don't delete this folder, e.g. https://github.com/conan-io/conan-center-index/blob/1ad155a70c49ec6c5c68c63d84bdc8f1913a40f7/recipes/aws-c-common/all/conanfile.py#L85